contracts/contracts/RequestManager.sol

```diff
@@ -198,14 +214,36 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
     /// Maps ERC20 token address to tokens
     mapping(address => Token) public tokens;
 
+    /// Compute the minimum liquidity provider fee that needs to be paid for a token transfer.
+    function minLpFee(
+        uint256 sourceChainId,
+        uint256 targetChainId,
+        address tokenAddress
+    ) public view returns (uint256) {
+        Token storage token = tokens[tokenAddress];
+        Chain storage sourceChain = chains[sourceChainId];
+        Chain storage targetChain = chains[targetChainId];
+        return
+            ((((1 - sourceChain.targetWeightPPM / 1_000_000) *
+                sourceChain.transferCost) /
+                1 ether +
+                ((targetChain.targetWeightPPM / 1_000_000) *
+                    targetChain.transferCost) /
+                1 ether) *
+                token.price *
+                (lpMargin + 100)) / 100;
```

Comment:
Am I assuming correctly that you consider the token price to be without decimals? The token price should be represented in the lowest unit as there is no float type in solidity.

If you divide the costs by 1 ether wouldnt the outcome be 0, if `transferCost` < 1 ether?

Reply:
I would probably consider lpMargin also to be PPM.